docs/PIPELINES-CONFIG.md

```diff
@@ -0,0 +1,269 @@
+```
+
+Take note of the public key, then use it to create the cosign secret under each of the deployment namespaces:
+
+```sh
+cat <<EOF | kubectl apply -f - >/dev/null
+apiVersion: v1
+data:
+    cosign.pub: <cosign_public_key>
+kind: Secret
+metadata:
+    labels:
+        app.kubernetes.io/instance: default
+        app.kubernetes.io/part-of: tekton-chains
+        operator.tekton.dev/operand-name: tektoncd-chains
+    name: cosign-pub
+    namespace: $APP_NAMESPACE
+type: Opaque
+EOF
+```
+
+#### Step 5: Setting up gitops authentication secret under deployment namespaces
+
+If you are using GitLab, you will need to create the GitLab authentication secret with your GitLab PAT as follows:
+
+```sh
+kubectl -n $APP_NAMESPACE create secret generic "gitlab-auth-secret" \
+    --from-literal=password=<gitlab_pat> \
+    --from-literal=username=oauth2 \
+    --type=kubernetes.io/basic-auth \
+    --dry-run=client -o yaml | kubectl -n $APP_NAMESPACE apply --filename - --overwrite=true >/dev/null
+```
+
+Otherwise, you will need to create a Git authentication secret with your PAT (e.g. GitHub PAT) as follows:
+
+```sh
+kubectl -n $APP_NAMESPACE create secret generic "gitops-auth-secret" \
+    --from-literal=password=<git_pat> \
+    --type=kubernetes.io/basic-auth \
+    --dry-run=client -o yaml | kubectl -n $APP_NAMESPACE apply --filename - --overwrite=true >/dev/null
+```
+
+#### Step 6: Setting up pipelines secret under deployment namespaces
+
+You will need to create a pipeline secret containing the webhook secret for the Git organization for deployments to have proper access via the tekton pipelines:
+
+```sh
+kubectl -n $APP_NAMESPACE create secret generic "pipelines-secret" \
+    --from-literal=webhook.secret=<webhook_secret> \
+    --dry-run=client -o yaml | kubectl -n $APP_NAMESPACE apply --filename - --overwrite=true >/dev/null
+```
+
+#### Step 7: Setting up quay image registry secret under deployment namespaces
+
+For accessing the quay image registry, you'll need to create a secret
+to store the docker config json file:
+
+```sh
+kubectl -n $APP_NAMESPACE create secret docker-registry "rhdh-image-registry-token" \
+    --from-file=.dockerconfigjson="<dockerconfig_jsonfile>" --dry-run=client -o yaml | \
+    kubectl -n $APP_NAMESPACE apply --filename - --overwrite=true >/dev/null
+```
+
+Then for both the `default` and `pipeline` service accounts under the deployment namespace, you will need to tie in the image registry secret you just created:
+
+```sh
+kubectl -n $APP_NAMESPACE patch serviceaccounts "<default|pipeline>" --patch "
+secrets:
+- name: rhdh-image-registry-token
+imagePullSecrets:
+- name: rhdh-image-registry-token
+" >/dev/null
+```
+
+#### Step 8: Updating Plugins
+
+To include the [Tekton plugins list](../dynamic-plugins/tekton-plugins.yaml) we need to edit the dynamic plugins ConfigMap that was created by the RHDH Operator:
```

Comment:
Out of curiosity, is there any difference if someone has installed RHDH through helm?

Reply:
> Out of curiosity, is there any difference if someone has installed RHDH through helm?

@thepetk The only difference is that they'll need to follow the 'Pre-Existing Instance' setup as installing through helm is separate from our installer and the environment variable `RHDH_PLUGINS` should be set to the different ConfigMap name for the dynamic plugins, normally is called `dynamic-plugins`.